scipy/signal/_signaltools.py

```diff
@@ -2339,8 +2339,8 @@ def hilbert2(x, N=None):
                          "two positive integers")
 
     Xf = sp_fft.fft2(x, N, axes=(0, 1))
-    h1 = np.zeros(N[0], 'd')
-    h2 = np.zeros(N[1], 'd')
+    h1 = np.zeros(N[0], dtype=Xf.dtype)
+    h2 = np.zeros(N[1], dtype=Xf.dtype)
```

Comment:
looks like it has core developer approval and a simple enough patch--should we worry that these two lines are not flushed by any tests?

![image](https://user-images.githubusercontent.com/7903078/148872656-dae6e3d1-1828-4bf7-b98d-fa0b1284f001.png)

Reply:
The only places `hilbert2` gets tested are with `assert_raises`:
```
(base) larsoner@bunk:~/python/scipy$ git grep hilbert2
...
scipy/signal/tests/test_signaltools.py:        assert_raises(ValueError, hilbert2, x)
scipy/signal/tests/test_signaltools.py:        assert_raises(ValueError, hilbert2, x)
scipy/signal/tests/test_signaltools.py:        assert_raises(ValueError, hilbert2, x, N=0)
scipy/signal/tests/test_signaltools.py:        assert_raises(ValueError, hilbert2, x, N=(2, 0))
scipy/signal/tests/test_signaltools.py:        assert_raises(ValueError, hilbert2, x, N=(2,))
```
@luigifcruz do you want to try adding some basic test in this PR? If not, I can push a quick commit to do it (assuming you have enabled "allow edits by maintainers")

Reply:
Sure @larsoner, I'll commit a test routine for this.

Reply:
Tests appear to be passing.